engine/src/widget.cpp

```diff
@@ -776,6 +777,41 @@ Boolean MCWidget::del(bool p_check_flag)
     return True;
 }
 
+void MCWidget::undo(Ustruct *us)
+{
+    MCControl::undo(us);
+
+    if (us->type == UT_REPLACE)
+    {
+        MCAutoValueRef t_rep;
+        t_rep.Give(m_rep);
+        m_rep = nullptr;
+        
+        MCNewAutoNameRef t_kind;
+        t_kind.Give(m_kind);
+        m_kind = nullptr;
+        
+        bind(*t_kind, *t_rep);
+    }
+}
+
+Boolean MCWidget::delforundo(bool p_check_flag)
+{
+    // Make the widget generate a rep.
+    if (m_widget != nil && m_rep == nullptr)
```

Comment:
I suggest saving the output of OnSave into a t_rep autocontainer - then only changing m_rep if `del` returns true - otherwise m_rep will be set whilst the control is bound.

Reply:
(Basically there should be no changed to the MCWidget's instance vars if del() returns false)